lib/create-testing-library-rule/detect-testing-library-utils.ts

```diff
@@ -769,20 +822,57 @@ export function detectTestingLibraryUtils<
     const isNodeComingFromTestingLibrary: IsNodeComingFromTestingLibraryFn = (
       node
     ) => {
-      const importNode = getImportedUtilSpecifier(node);
+      const importNode = getTestingLibraryImportedUtilSpecifier(node);
 
       if (!importNode) {
         return false;
       }
 
+      const referenceNode = getReferenceNode(node);
+      const referenceNodeIdentifier = getPropertyIdentifierNode(referenceNode);
+      if (!referenceNodeIdentifier) {
+        return false;
+      }
+
+      const importDeclaration = (() => {
+        if (isImportDeclaration(importNode.parent)) {
+          return importNode.parent;
+        }
+
+        const variableDeclarator = findClosestVariableDeclaratorNode(
+          importNode
+        );
+
+        if (isCallExpression(variableDeclarator?.init)) {
+          return variableDeclarator?.init;
+        }
+
+        return undefined;
+      })();
+
+      if (!importDeclaration) {
+        return false;
+      }
+
+      const importDeclarationName = getImportModuleName(importDeclaration);
+      if (!importDeclarationName) {
+        return false;
+      }
+
       const identifierName: string | undefined = getPropertyIdentifierNode(node)
         ?.name;
 
       if (!identifierName) {
         return false;
       }
 
-      return hasImportMatch(importNode, identifierName);
+      const hasImportElementMatch = hasImportMatch(importNode, identifierName);
+      const hasImportModuleMatch =
+        /testing-library/g.test(importDeclarationName) ||
+        (typeof customModuleSetting === 'string' &&
+          importDeclarationName.endsWith(customModuleSetting));
```

Comment:
This is another of the issues I found and I needed to fix already. Without this check, an identifier matching some Testing Library util name but imported from an unrelated module could be assumed as coming from Testing Library package.

```js
import { act as renamedAct } from '@testing-library/react'
import { act } from 'somewhere-else'

// this should not be reported since it's an `act` but not coming from Testing Library,
// but without previous fix it was reported since it found an import matching the specifier.
act(() => {})
```